plc4j/drivers/opcua/src/main/java/org/apache/plc4x/java/opcua/connection/OpcuaTcpPlcConnection.java

```diff
@@ -346,7 +346,7 @@ public void unregister(PlcConsumerRegistration registration) {
             for(int counter = 0; counter < readValueIds.size(); counter++){
                 PlcResponseCode resultCode = PlcResponseCode.OK;
                 BaseDefaultFieldItem stringItem = null;
-                if(readValues == null || readValues.size() <= counter || readValues.get(counter).getStatusCode() != StatusCode.GOOD){
+                if(readValues == null || counter > readValues.size() || !readValues.get(counter).getStatusCode().isGood()){
```

Comment:
Correct but it has to be `counter >= readValues.size()` or?

Reply:
Actually I think you are right as if it's equal to size it must produce an array index out of bounds exception. Stefan, could you please re evaluate that?

Reply:
you are correct, will fix that.. :)